Add docs about brew upgrade HZ-618 (#27)
hazelcast/homebrew-hz: README.md

# Homebrew Tap Repository for Hazelcast Platform

> `hz` is a tool which allows users to install & run [Hazelcast IMDG](https://hazelcast.org/imdg/) on a local environment.

Reviewer: IMDG?

Author: hmm, I don't follow, should I use `Hazelcast Platform` instead?

Author: changed: https://github.com/hazelcast/homebrew-hz/pull/27/commits/138cc6e9f9dc3a74f033fb613c85fa1e881c1449

> `hzcloud` is a command line tool to make actions on Hazelcast Cloud easily.

Reviewer: 'Make actions' isn't helpful.

Maybe 'for managing Hazelcast Cloud clusters'?

Author: Changed: https://github.com/hazelcast/homebrew-hz/pull/27/commits/138cc6e9f9dc3a74f033fb613c85fa1e881c1449

## About

> This repository contains the Homebrew formulas for `hazelcast`, `hazelcast-enterprise` and  `hzcloud`.

Reviewer: I think this should be the first sentence before we start explaining what `hz` and `hc-cli` are.

We should also provide a brief description of each one and then describe what CLI tools come with each (`hz` or `hzcloud`).

Author: Changed: https://github.com/hazelcast/homebrew-hz/pull/27/commits/138cc6e9f9dc3a74f033fb613c85fa1e881c1449

See [Hazelcast Packaging](https://github.com/hazelcast/hazelcast-packaging) for the source code.

See [Hazelcast Cloud Command Line](https://github.com/hazelcast/hazelcast-cloud-cli) for the source code

## How to install

> ### Add tap repository

Reviewer: ```suggestion
### Add Tap Repository
```

    brew tap hazelcast/hz

### Install Hazelcast

Install the community version of Hazelcast

    brew install hazelcast 

Upgrade the community version of Hazelcast

    brew upgrade hazelcast 

Install particular minor version of Hazelcast

    brew install hazelcast-5.1

Install particular version of Hazelcast

    brew install hazelcast@5.0.1

> NOTE: Upgrade of a particular version is not supported by brew. You need to uninstall the old package first 

Reviewer: ```suggestion
NOTE: To upgrade to a particular version, uninstall the old package first 
```

> and then install `hazelcast` or `hazelcast-x.y` instead

Reviewer: ```suggestion
and then install `hazelcast` for the latest version or `hazelcast-x.y` for another version.
```

Install snapshot version of Hazelcast

    brew install hazelcast@5.1.snapshot

Install Hazelcast Enterprise edition (you need a license key to run it,
see [Installing a License Key](https://docs.hazelcast.com/hazelcast/latest/getting-started/get-started-enterprise#installing-a-license-key)

    brew install hazelcast-enterprise

Install without Java dependency (you already have installed Java in 
another way)

    brew install hazelcast --without-openjdk

### Configuration

> Starting from the 5.1 version configuration files are preserved between upgrades and stored in

Reviewer: ```suggestion
Starting from version 5.1, configuration files are preserved between upgrades and stored in
```

    $(brew --prefix)/etc/hazelcast

(usually `/usr/local/etc/hazelcast` or `/opt/homebrew/etc/hazelcast`)

### Install Hazelcast CLoud CLI

    brew install hzcloud

> ### To clean up everything (may come in handy):

Reviewer: ```suggestion
### Clean Up
```

>     brew uninstall hazelcast

Reviewer: Should we add `\` to the end of these lines so that copy/paste works?

Author: They're all non-interactive commands so new lines should be fine
